### ISSUE TITLE
HD GUI 2 TODO

### ISSUE THREAD

**Issue by PiKeyAr** [PiKeyAr/sadx_dreamcast#141]
Updated TODO list with latest contributions

**General**
- [x] Company logos in credits
- [ ] Optional: Make the frame in J and E level objectives more consistent (@DisapprovingOwl and/or @SPEEPSHighway)

**Maps**
- [x] Station Square
- [x] Egg Carrier Hall
- [x] Egg Carrier Outside 1
- [x] Egg Carrier Outside 2
- [x] Mystic Ruins main area
- [x] Mystic Ruins Emerald Altar
- [x] Mystic Ruins Jungle
- [x] Past: Echidna City
- [x] Past: Emerald Altar
- [ ] Japanese maps

**Currently unsupported (not scaled by the Mod Loader)**
- [ ] Subtitle font
- [ ] Mini Game Collection main menu 
- [ ] Mini Game Collection tutorial 
- [ ] Mini Game Collection pause 
- [ ] Chao fonts 
- [ ] Chao button prompts (removed in DC_ChaoGardens)
- [ ] Name Machine UI 
- [ ] Black Market UI 
- [ ] Chao Race Entry UI 
- [ ] Chao Race HUD

**Comment by PiKeyAr** [comment]
HD GUI 2 has been released. I'm going to close this until the Mod Loader is updated with support for stuff that wasn't included.

**Comment by DisapprovingOwl** [comment]
Text errrors/alignment fixes for the HD GUI versions of the stage mission cards, as discussed over on Twitter.

[SA_MISSIONCARDS_FIXED.zip](https://github.com/PiKeyAr/sadx_dreamcast/files/1675623/SA_MISSIONCARDS_FIXED.zip)

Full list of changes:

**GENERAL AMY MISSION CARD CHANGES**
- Corrected spelling of "balloon" on all included cards
- Fixed balloon handle in image on all included cards
- Realigned text on all included cards for better consistency between Amy's missions

**MISSION_A_BALRING_E**: 
- Recoloured "50" red to match other "collect 50 rings" missions

**GENERAL KNUCKLES MISSION CARD CHANGES**
- Realigned text for better consistency between Knuckles' missions

**GENERAL SONIC MISSION CARD CHANGES**
- Realigned text for better consistency between Sonic's timed missions

**MISSION_S_BOX2MIN_E**
- Replaced "2" to remove weird dark outline

**MISSION_S_BOX4MIN_E**
- Replaced "4" with one using correct font/weighting, added period to end of sentence to match other timed missions

**MISSION_S_BOX25MIN_E**
- Removed red streak from bottom of "y" in "Destroy"

**MISSION_T_RINGEGG_E**
- Replaced "50" to remove weird shadow effect not present on any other mission card

**Comment by DisapprovingOwl** [comment]
@PiKeyAr - Huge update in the attached ZIP archive - this makes changes both minor and major to all Japanese and English cards to fix more issues and increase quality/consistency further where possible. It also adds a white border to all English cards as requested on Twitter.

[SA_HD-MISSION-CARDS_2018-02-10.zip](https://github.com/PiKeyAr/sadx_dreamcast/files/1713581/SA_HD-MISSION-CARDS_2018-02-10.zip)

Below is a quick overview of the changes made for each language:

**English:**
- [x] Realigned all cards so that their horizontal alignment is consistent.
- [x] Added white border to all cards, mimicking the one from the Japanese cards.
- [x] Manually tweaked some elements to clean up upscaling artefacts, including the capsule's base and the areas where Big's fish/Sonic's spine/the Egg Carrier/Robotniks Egg Mobile overlap the top of the card.
- [x] Realigned capsule on Tails' mission card for consistency with Sonic's.
- [x] Substituted all images of E-103/E-104/E-105 with high-quality official renders where appropriate. Minor differences to E-105 due to the HQ render featuring the floating objects which surround him, which weren't present on the otherwise identical render used on the original card. E-104's gun now correctly overlaps the top of the card.
- [x] Corrected text alignment where appropriate on new E-103/E-104/E-105 cards.
- [x] Cleaned up artefacts visible in the black border on the C-Rank cards for Sonic's Red Mountain/Tails' Speed Highway.
- [x] Restored the missile's lights on Tails' C-Rank Speed Highway card - these were visible on the Japanese/French/Spanish cards, but absent from the English/German cards.

**Japanese:**
- [x] Realigned all cards so that their horizontal alignment is consistent.
- [x] Cleaned up white border for consistency between cards.
- [x] Performed same artefact cleanup as English cards for the capsule's base and the areas where Big's fish/Sonic's spine/the Egg Carrier/Robotniks Egg Mobile overlap the top of the card.
- [x] Realigned capsule on Tails' mission card for consistency with Sonic's.
- [x] Performed same substitution of E-103/E-104/E-105 with high-quality official renders as for the English cards - the minor differences noted for the English cards also apply here. Have kept E-105's position higher up than the English cards per the original Japanese cards.
- [x] Cleaned up artefacts visible in the black border on a large number of cards.
- [x] Realigned capsule on Tails' mission card for consistency with Sonic's.
- [x] Added missing glow to text on Gamma's C-Rank Final Egg/Hot Shelter cards and Sonic's C-Rank Lost World card.
- [x] Added missing purple sky gradient present to Tails' B and A-Rank Speed Highway cards - this was present on some language versions of the card but not others; I've migrated it over as it looks more consistent with the purple sky seen in all language versions of the C-Rank mission.

Have not touched the Goal Ring cards yet as they look like they'll need redoing from scratch - the fonts used aren't correct and the Goal Ring image could do with improving. I've been having trouble finding a high-quality image of the SA2-styled Goal Ring to use - do you know if one exists, or if not, whether anyone would be kind enough to supply an appropriate render for use on the updated cards?

**Comment by DisapprovingOwl** [comment]
@PiKeyAr - New ZIP attached updating a good chunk of the English cards to incorporate the lovely manual paint edits from SPEEPSHighway's HD Japanese cards. Only cards which have been updated since the previous version are included in the ZIP:

[SA_HD-MISSION-CARDS_2018-02-11.zip](https://github.com/PiKeyAr/sadx_dreamcast/files/1714326/SA_HD-MISSION-CARDS_2018-02-11.zip)

This incorporates the following edits to improve image quality:

- [x] Balloon image replaced on all Amy's cards.
- [x] Emerald shards image replaced on all Knuckles' cards.
- [x] Capsule image replaced on all appropriate Sonic/Tails cards.
- [x] Egg Carrier image replaced on Sonic's C-Rank Red Mountain card.
- [x] Tails image replaced on Sonic's C-Rank Emerald Coast card.
- [x] Robotnik images replaced on Tails' B and A-Rank Speed Highway cards. These incorporate SPEEPSHighway's detail edits to Robotnik and the Egg Mobile while preserving the purple sky from the original English card.
- [x] Sonic images replaced on Tails' other B and A-Rank cards.
- [x] Missile image replaced on Tails' C-Rank Speed Highway card.

Also, flagged this up on Twitter but just in case you miss that, noticed a slight error with the new Japanese version of MISSION_T_MISS - the card's contents are misaligned in comparison to the borders, so the black border ends up thinner on the left than on the right.

**Comment by DisapprovingOwl** [comment]
Small error fixed on the above - the manually-repainted Sonic image from the new Japanese cards has a small rectangle of the background missing above and to the right of Sonic's head - this was copied over to the English cards when making the above fixes. 

Have updated the two English cards affected (**MISSION_T_RINGSONIC_E** and **MISSION_T_FASTSONIC_E**) in the attached ZIP to add back in the missing area of background - please use these instead of the versions included in the ZIP in the previous post above!

[SA_HD-MISSION-CARDS_SONICFIX_2018-02-11.zip](https://github.com/PiKeyAr/sadx_dreamcast/files/1714937/SA_HD-MISSION-CARDS_SONICFIX_2018-02-11.zip)

Goal ring cards coming next. = )

**Comment by DisapprovingOwl** [comment]
@PiKeyAr - English cards for the Goal Ring mod now available in the attached .ZIP:

[SA_HD-MISSION-CARDS_GOALRING_2018-02-11.zip](https://github.com/PiKeyAr/sadx_dreamcast/files/1714957/SA_HD-MISSION-CARDS_GOALRING_2018-02-11.zip)

These match the font/positioning/sizing of text to the other English cards, mostly using the Capsule cards they replace for reference. They also replace the background with a higher quality version sourced from the other updated English mission cards, and the Goal Ring itself with a higher-quality image based on the PNG you kindly provided.

Hope they're OK - let me know if any tweaks/changes are needed. = )

**Comment by cheatfreak47** [comment]
Are the project or resource files for HDGUI avaliable? If they are i might be interested in working on versions of all the various screens that have button prompts, versions of the textures with all the different possible console button prompts. I'd likely use Xelu's free and open- royalty free graphics for button prompts if I did that since I think they'd fit in quite well.

**Comment by Sonikko** [comment]
@DisapprovingOwl Hey, are you on Discord by any chance? I've been meaning to remake the Level Cards in HD aswell, we might work together!

**Comment by DisapprovingOwl** [comment]
@Sonikko I'm not at the moment, but would be willing to get up and running to collaborate if that'd help!

If you're talking about what I think you are - proper HD remakes of the card images/backgrounds - then those would be a great asset and definitely worth integrating IMO. At the moment the only true HD bits we've been able to incorporate are the E-100 series droids, due to their official renders being available for use. And even on those ones, the backgrounds they sit on are still upscaled from the originals - it'd be great if we could get suitable screenshots of the appropriate areas from as closely matched angles as possible to replace them!

**Comment by Sonikko** [comment]
Oh lol, I thought you had re-rendered them.
If you look closely, the backgrounds aren't ingame screenshots, they feature shadows and ambient occlusion which are not possible ingame. I'd be glad to do those in Blender, it should be fairly easy to do so.
Where are you getting the renders exactly? Are more of them available somewhere?

**Comment by DisapprovingOwl** [comment]
Ah, sounds like you have a better eye for details in those backgrounds than I do, haha. Glad to hear you should be able to do higher-quality versions of those in Blender - that'd be a great help.

In terms of foreground objects, unfortunately the E-100 series are the only ones that I'm aware publically-available renders exist for, otherwise I'd have tried to integrate more. I found those over at the Sonic Retro wiki's SA art archive: http://info.sonicretro.org/Category:Artwork_for_Sonic_Adventure

The E-105 render is actually slightly different to the one used in-game as it has three of the missiles floating around him (including one directly in front, which was annoying as it wasn't possible to paint it out convincingly - hence why the missiles appear on the current HD English card set).

**Comment by DisapprovingOwl** [comment]
@PiKeyAr - In the meantime, here's an updated copy of all English cards, including the extra ones for the Goal Ring mod:

[SA_HD-MISSION-CARDS_E_2018-02-18.zip](https://github.com/PiKeyAr/sadx_dreamcast/files/1734776/SA_HD-MISSION-CARDS_E_2018-02-18.zip)

In this update, I've given the whole set a consistency pass to bring all the cards much closer in line with SPEEPS' current Japanese set. Tweaks include changing the white border to match, integrating as much of the imagery from SPEEPS' cards as possible to ensure consistency, and incorporating the glows/outlines that were present on objects which overlap the edges of the cards in SPEEPS' set. A similar glow has also been added to the updated E-104 background for consistency. The more purple-looking sky in the vs. Robotnik cards has been kept as that's how it was in the original English cards, but other than that things should be about as consistent as we can get them for the time being!

The versions of the Goal Ring and E-103/4/5 backgrounds that I sent SPEEPS for integration into the Japanese set have already had any extra tweaks I made here applied to them, so in theory they should be completely consistent with this set once they're in.

Hope that's all OK - if you spot anything which needs fixing/changing, just let me know! = )

**Comment by SPEEPSHighway** [comment]
Tried to send this earlier on Twitter but it apparently hates Google Drive links. Here's the Goal Rings (Text from Windii):
[GoalRing.zip](https://github.com/PiKeyAr/sadx_dreamcast/files/1734781/GoalRing.zip)

**Comment by DisapprovingOwl** [comment]
Those are looking awesome, @SPEEPSHighway! Great to see proper Japanese language versions of those extra cards. = )

**Comment by PiKeyAr** [comment]
Just pushed the latest update that includes slightly edited maps by @Sonikko. The recreated maps looked very similar to the ones we currently have, so we decided to keep the current ones. However, we also removed the background to make these maps work better in widescreen. 

The new update also includes upscaled Japanese maps that have had the basic work done, but they would benefit from text recreation. If @SPEEPSHighway or anyone else wants to do it, feel free to pick up [this archive](https://dcmods.unreliable.network/owncloud/data/PiKeyAr/files/Various/hdgui_maps.7z), which contains current English and Japanese maps.
